src/main/java/org/apache/sling/i18n/impl/JcrResourceBundleProvider.java

```diff
@@ -264,7 +247,7 @@ private void onChange(final ChangeStatus status, final ResourceChange change)
             for (final String root : languageRootPaths) {
                 if (change.getPath().startsWith(root)) {
                     // figure out which JcrResourceBundles from the cached ones is affected
-                    for (JcrResourceBundle bundle : resourceBundleCache.values()) {
+                    for (JcrResourceBundle bundle : resourceBundleRegistry.getRBs()) {
```

Comment:
I would stick with `getResourceBundles()`. It is longer, but it also reads better IMHO.

Reply:
Ok for me.